### ISSUE TITLE
[test-triage] chip_sw_coremark

### ISSUE THREAD

**Issue by johngt** [lowRISC/opentitan#14330]
### Hierarchy of regression failure

Chip Level

### Failure Description

```
ERROR: Skipping '//sw/device/benchmarks/coremark/coremark_top_earlgrey:1_sim_dv': no such package 'sw/device/benchmarks/coremark/coremark_top_earlgrey': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
   - /workspace/mnt/repo_top/sw/device/benchmarks/coremark/coremark_top_earlgrey
  WARNING: Target pattern parsing failed.
  ERROR: no such package 'sw/device/benchmarks/coremark/coremark_top_earlgrey': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
   - /workspace/mnt/repo_top/sw/device/benchmarks/coremark/coremark_top_earlgrey
  INFO: Elapsed time: 19.608s
  INFO: 0 processes.
  FAILED: Build did NOT complete successfully (0 packages loaded)
  FAILED: Build did NOT complete successfully (0 packages loaded)
  make: *** [/workspace/mnt/repo_top/hw/dv/tools/dvsim/sim.mk:73: sw_build] Error 1
```

### Steps to Reproduce

Observed in all nightlies since at least 12 August.
[cab5ffb90](https://github.com/lowrisc/opentitan/tree/cab5ffb9055258dcad355d60ac2c655519fbc000)
Build seed 3773991536, VCS

### Tests with similar or related failures

NA

**Comment by timothytrippel** [comment]
This actually has not worked since the meson-->bazel transition (see #12504).
cc: @arunthomas

**Comment by timothytrippel** [comment]
Also, see #12671

**Comment by johngt** [comment]
Given that this item is being worked on by @arunthomas / @cfrantz but has not been touched for some time we will unassign from @engdoreis. @timothytrippel if you want to leave this assigned to someone from your side please mark accordingly but otherwise we will observe/track for now.

**Comment by timothytrippel** [comment]
SG, reassigning to myself / @arunthomas for now.

**Comment by arunthomas** [comment]
Bazel build support for CoreMark was merged yesterday. @timothytrippel  will be working on adding CoreMark to the nightly regressions.

**Comment by tjaychen** [comment]
actually, do we think it's important to have this as a v2 test? 
This doesn't really check for anything specific functional changes..

in theory we would use this to know our architecture is too slow and requires big changes, but we are far beyond that point now.

**Comment by timothytrippel** [comment]
Agreed with @tjaychen , but since this took literally 30s to fix, I fixed it hehe (in #16091). Sorry for the delay.

**Comment by tjaychen** [comment]
even better hehe.

**Comment by engdoreis** [comment]
I'm reopening this due to the failured in the latest regression and I'll close it when it pass in 3 consecutive regressions.

**Comment by johngt** [comment]
@timothytrippel : just highlighting that this has been reopened.
It looks like it was just a timeout issue. 
Not sure how long coremark will need to run but the default timeout may need to be upped. @engdoreis - possible to replicate on local system and see how long the task takes and then set timeout accordingly?

```
Job chip_earlgrey_asic-sim-vcs_run_default killed due to: Exit reason: User job exceeded runlimit: User job timed out has 6 failures:

Test chip_sw_coremark has 3 failures.
0.chip_sw_coremark.3709997532
Log /container/opentitan-public/scratch/os_regression/chip_earlgrey_asic-sim-vcs/0.chip_sw_coremark/latest/run.log

  Job ID: smart:81bccfa8-4935-40f9-8af6-3fd285e3976b
```

**Comment by weicaiyang** [comment]
@timothytrippel This test still fails with this error. Can you take another look?

> 0.chip_sw_coremark.1083389417
Line 486, in log /container/opentitan-public/scratch/os_regression/chip_earlgrey_asic-sim-vcs/0.chip_sw_coremark/latest/run.log

  UVM_ERROR @ 22018.182637 us: (chip_sw_base_vseq.sv:237) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.chip_sw_base_vseq] SW TEST TIMED OUT. STATE: SwTestStatusInTest, TIMEOUT = 22000000 ns

**Comment by tjaychen** [comment]
i'm honestly not sure this test can finish in a reasonable DV window... haha.
I hope the test hasn't locked up or something...

**Comment by johngt** [comment]
The timeout was upped to 3hrs according to the PR that @timothytrippel set. This seems a very long time for coremark to need to run. Might be worth trying this in a local environment and check that there are no settings that are causing this to run an infinite amount of times and waiting for user input etc.

**Comment by timothytrippel** [comment]
#16623 should finally resolve this. Confirmed locally on my workstation.

**Comment by engdoreis** [comment]
It's passing on the Regression.

**Comment by engdoreis** [comment]
@GregAC are you looking into this? If so can I assign it to you?

**Comment by johngt** [comment]
@moidx - I think this was merged in by @timothytrippel so may already be resolved?

**Comment by vogelpi** [comment]
I recently ran the test locally, the output is as follows:
```
UVM_INFO @ 2294.065684 us: (sw_logger_if.sv:524) [test_rom_sim_dv(sw/device/lib/testing/test_framework/status.c:40)] test_status_set to 0xb090
UVM_INFO @ 2294.335684 us: (sw_test_status_if.sv:61) [tb.u_sim_sram.u_sim_sram_if.u_sw_test_status_if] SW test transitioned to SwTestStatusInBootRom.
UVM_INFO @ 2455.445684 us: (sw_logger_if.sv:524) [test_rom_sim_dv(sw/device/lib/testing/test_rom/test_rom.c:158)] kChipInfo: scm_revision=54697461
UVM_INFO @ 2461.585684 us: (sw_logger_if.sv:524) [test_rom_sim_dv(sw/device/lib/testing/test_rom/test_rom.c:230)] Test ROM complete, jumping to flash (addr: 20000480)!
UVM_INFO @ 2467.745684 us: (sw_logger_if.sv:524) [coremark_test_sim_dv(w/device/lib/testing/test_framework/status.c:40)] test_status_set to 0x4354
UVM_INFO @ 2467.975684 us: (sw_test_status_if.sv:61) [tb.u_sim_sram.u_sim_sram_if.u_sw_test_status_if] SW test transitioned to SwTestStatusInTest.
UVM_INFO @ 2467.975684 us: (chip_sw_uart_tx_vseq.sv:10) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.chip_sw_uart_tx_vseq] Configuring and connecting UART agent to UART 0 ...
UVM_INFO @ 65126.905525 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] 2K performance run parameters for coremark.
UVM_INFO @ 65385.930613 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] CoreMark Size    : 666
UVM_INFO @ 65687.911285 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Total ticks      : 6193300
UVM_INFO @ 65934.895477 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Total time (secs): 12
UVM_INFO @ 66167.880565 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Iterations/Sec   : 1
UVM_INFO @ 66414.864757 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Iterations       : 12
UVM_INFO @ 67857.772405 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Compiler version : GCClowRISC Clang 16.0.2 (https://github.com/lowRISC/llvm-project.git d213f6b2e0bcc561930538ef47b3b5cc900b5b17)
UVM_INFO @ 68533.770805 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Compiler flags   : Please put compiler flags here (e.g. -o3)
UVM_INFO @ 68811.753013 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Memory location  : STACK
UVM_INFO @ 69104.734261 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] seedcrc          : 0xe9f5
UVM_INFO @ 69397.715509 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] [0]crclist       : 0xe714
UVM_INFO @ 69692.696629 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] [0]crcmatrix     : 0x1fd7
UVM_INFO @ 69986.677813 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] [0]crcstate      : 0x8e3a
UVM_INFO @ 70279.659061 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] [0]crcfinal      : 0x755b
UVM_INFO @ 71077.649653 us: (uart_logger.sv:90) uvm_test_top.env.m_uart_agent0.m_logger [uvm_test_top.env.m_uart_agent0.m_logger] Correct operation validated. See README.md for run and reporting rules.
UVM_INFO @ 71078.605684 us: (sw_logger_if.sv:524) [coremark_test_sim_dv(w/device/lib/testing/test_framework/status.c:28)] PASS!
UVM_INFO @ 71078.865684 us: (sw_test_status_if.sv:61) [tb.u_sim_sram.u_sim_sram_if.u_sw_test_status_if] SW test transitioned to SwTestStatusPassed.
UVM_INFO @ 71078.865684 us: (sw_test_status_if.sv:82) [tb.u_sim_sram.u_sim_sram_if.u_sw_test_status_if] ==== SW TEST PASSED ====
UVM_INFO @ 71078.865684 us: (chip_sw_base_vseq.sv:302) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.chip_sw_uart_tx_vseq] SW TEST PASSED!
```
@GregAC , can you check if these numbers make sense?

**Comment by vogelpi** [comment]
I think the test failures are due to the test timing out. The last passing test in the nightlies (June 28) took more than 3.5 hours where as the timeout occurs at 4 hours currently. I can file a PR to further increase the timeout?

**Comment by andreaskurth** [comment]
@vogelpi to please reduce iterations so that it fits the timeout and calculate CoreMark per MHz

**Comment by vogelpi** [comment]
I did some more digging. The current CoreMark/MHz is ~1.93 which is inline with what [Greg reported in May 2023](https://github.com/lowRISC/opentitan/issues/17370#issuecomment-1453324348).

We should not lower the number of iterations because CoreMark expects to run for at least 10 seconds and reports an error otherwise. What we can do though is increasing the timeout a bit (we recently increased the number of iterations targeting FPGA and silicon but forgot to increase the DV timeout). I'll take care of this and also reduce the number of seeds for this test. Because it's pretty deterministic anyway.
